### DISCUSSION TITLE
Hex color

### DISCUSSION THREAD

**Discussion by rexkenley** [microsoft/fluentui#31171] [Q&A] [answered]
![image](https://github.com/microsoft/griffel/assets/10548251/4566b84c-55b7-46a7-8b36-e7fc23ef11b7)

Console
`Primary : ___1vvfegh_0000000 f16muhyy`

How do I get the hex color code?

**Comment by layershifter** [comment]
TL;DR https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle

_FYI That's not related to Griffel as it's just emits CSS and provides classes to apply._

On Fluent side, it's expected to always use CSS variables i.e. `tokens.SMTH`.

**Reply by rexkenley** [reply]
Thanks for the reply. But what am I doing wrong?

![image](https://github.com/microsoft/griffel/assets/10548251/f3a9e4db-5858-4ea8-bf77-a4d71ab9330b)

**Reply by layershifter** [reply]
Check examples on MDN, https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle#javascript. `window.getComputedStyle()` accepts an `Element` as a param.

**Reply by rexkenley** [reply]
Is there a way to get the hex color without the need of an element? 

I originally put the theme in a context object to get the hex value. That theme is the same one that I provide to the FluentProvider. I was hoping to be able to get the hex value through child inheritance instead. 

![image](https://github.com/microsoft/griffel/assets/10548251/1b1c62eb-2b56-46fe-b0ea-30954a664c0e)
![image](https://github.com/microsoft/griffel/assets/10548251/466e9099-7257-446d-8cb0-4226d848197d)

I need the hex value because I am using a mui component.
https://mui.com/material-ui/customization/theming/#themeprovider

**Reply by layershifter** [reply]
Not sure that I follow, do you mind to create an example on Stackblitz?

**Reply by rexkenley** [reply]
Here you go. 
https://stackblitz.com/edit/jzxava-vz54gd

If you look at the console, the webLightTheme.colorBrandForeground1 returns #0f6cbd, which is what I need.

So in essence, how do I go from tokens.colorBrandForeground1 to the #0f6cbd value? I am supposed to be able to get that value somehow since the FluentProvider is the parent element, right?

**Reply by layershifter** [reply, marked as answer]
The proper solution would be to require something like `BridgeProvider` (example https://stackblitz.com/edit/jzxava-j8xopl) or create a custom React context and pass down `theme` through it.

We intentionally removed all ways to access exact values to ensure that they are never referenced in styles and won't break theming (#21257).

---

Technically, you can still do `.getComputedStyle()`, but it's fragile as it won't react on theme changes, https://stackblitz.com/edit/jzxava-wzh5gr.

**Reply by rexkenley** [reply]
Thanks @layershifter 
![image](https://github.com/microsoft/fluentui/assets/10548251/f0ab636a-60e2-4839-afdc-1dbbd5d71ada)
